### PR TITLE
Remove legacy toolchain support from jvm tasks

### PR DESCRIPTION
**Pull request by bmuskalla**
Removes toolchain support from
* `Javadoc`
* `JavaCompile`
* `GroovyCompile`

### CODE REVIEW

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Comment by bmuskalla** [comment]
@ljacomet Do you have *any* pointer on how these changes could cause the issues on CI? Especially `> Could not set unknown property 'languageVersion' for task ':test' of type org.gradle.api.tasks.testing.Test.`. Why does it suddenly not recognize `languageVersion` being part of the spec object that we pass into the toolchain services. Example test: https://builds.gradle.org/buildConfiguration/Gradle_Check_Quick_1_bucket39/40170844?showLog=40170844_7383_1927.7365.7383&logFilter=debug

**Comment by bmuskalla** [comment]
@ljacomet this should be good for review

**Review by ljacomet — APPROVED** [review]
LGTM
Minor nitpicks:

File: subprojects/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java [hunk 1]
```diff
@@ -196,45 +196,33 @@ private void executeExternalJavadoc(StandardJavadocDocletOptions options) {
...
 
-    /**
-     * Returns the tool chain that will be used to generate the Javadoc.
-     */
     @Inject
-    @Deprecated
-    public JavaToolChain getToolChain() {
-        // Implementation is generated
+    protected JavaToolchainService getJavaToolchainService() {
         throw new UnsupportedOperationException();
```
[the remark this answers is not in the document]
Comment (bmuskalla) [reply subprojects/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java:216]: Do you think this is neccessary? We use this pattern for injection all over the place (see also all the methods at the end of `Javadoc`

[the remark this answers is not in the document]
Comment (ljacomet) [reply subprojects/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java:216]: I find it useful yes. But I do not think it is mandatory.
Learning the pattern with that comment in is much simpler than without.

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Pull request merged**